src/include/duckdb/common/radix_partitioning.hpp

```diff
@@ -9,52 +9,61 @@
 #pragma once
 
 #include "duckdb/common/fast_mem.hpp"
-#include "duckdb/common/types/partitioned_column_data.hpp"
+#include "duckdb/common/types/column/partitioned_column_data.hpp"
+#include "duckdb/common/types/row/partitioned_tuple_data.hpp"
 
 namespace duckdb {
 
 class BufferManager;
-class RowLayout;
-class RowDataCollection;
 class Vector;
 struct UnifiedVectorFormat;
 struct SelectionVector;
 
-//! Templated radix partitioning constants, can be templated to the number of radix bits
-template <idx_t radix_bits>
-struct RadixPartitioningConstants {
+//! Generic radix partitioning functions
+struct RadixPartitioning {
 public:
-	static constexpr const idx_t NUM_RADIX_BITS = radix_bits;
-	static constexpr const idx_t NUM_PARTITIONS = (idx_t)1 << NUM_RADIX_BITS;
-	static constexpr const idx_t TMP_BUF_SIZE = 8;
+	static inline constexpr idx_t NumberOfPartitions(const idx_t &radix_bits) {
```

Comment:
I've gotten used to using a const reference everywhere, because clang-tidy will complain about this when the arguments to a function are not trivially copyable, like `vector`. Of course, `idx_t` is trivially copyable so this is not necessary, but it does not hurt, either.